testsuite/pangu-sdk/src/lib.rs

```diff
@@ -0,0 +1,829 @@
+            .arg("-C")
+            .arg(&pangu_dir)
+            .arg("run")
+            .arg("python")
+            .arg(format!("{}/pangu.py", pangu_dir))
+            .arg("testnet")
+            .arg("get")
+            .arg("-o")
+            .arg("json")
+            .stderr(Stdio::piped())
+            .output()
+            .expect("Failed to execute command");
+
+        let stdout = String::from_utf8_lossy(&command_output.stdout).to_string();
+        let stderr = String::from_utf8_lossy(&command_output.stderr).to_string();
+
+        if command_output.status.success() {
+            let testnet_headers: TestnetHeaders = serde_json::from_str(&stdout).map_err(|err| {
+                eprintln!("Failed to deserialize JSON: {}", err);
+                format!("Failed to deserialize JSON: {}", err)
+            })?;
+            Ok(testnet_headers)
+        } else {
+            Err(format!("Failed to get testnets: {}", stderr))
+        }
+    }
+
+    fn node_action_helper(
+        action: &str,
+        testnet_name: &str,
+        node_name: &str,
+    ) -> Result<String, String> {
+        let pangu_dir = Self::pangu_directory();
+        let command_output = Command::new("poetry")
+            .arg("-C")
+            .arg(&pangu_dir)
+            .arg("run")
+            .arg("python")
+            .arg(format!("{}/pangu.py", pangu_dir))
+            .arg("node")
+            .arg(action)
+            .arg(testnet_name)
+            .arg(node_name)
+            .stderr(Stdio::piped())
+            .output()
+            .expect("Failed to execute command");
+
+        let stderr = String::from_utf8_lossy(&command_output.stderr).to_string();
+        let stdout = String::from_utf8_lossy(&command_output.stdout).to_string();
+
+        if command_output.status.success() {
+            Ok(format!(
+                "Initiated to {} node \"{}\" in testnet \"{}\".",
+                action, node_name, testnet_name
+            ))
+        } else {
+            Err(format!(
+                "Failed to {} node \"{}\" in testnet \"{}\": \n{}, \n{}",
+                action, node_name, testnet_name, stderr, stdout
+            ))
+        }
+    }
+
+    fn create_pangu_node_config(
+        blueprint: &BlueprintCollection,
+    ) -> Result<NamedTempFile, Box<dyn std::error::Error>> {
+        let yaml_data = serde_yaml::to_string(blueprint)?;
+        let mut temp_file = NamedTempFile::new()?;
+        temp_file.write_all(yaml_data.as_bytes())?;
+        Ok(temp_file)
+    }
+
+    fn pangu_directory() -> String {
+        if let Ok(pangu_dir) = env::var("PANGU_DIR") {
+            pangu_dir
+        } else {
+            panic!("Failed to retrieve PANGU_DIR environment variable.");
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_create_testnet() {
+        env::set_var(
+            "PANGU_DIR",
+            "/Users/olsenbudanur/Desktop/aptos-repos/aptos-core/testsuite",
```

Comment:
maybe just assert that `PANGU_DIR` env var is set? or derive it as a relative path from this dir? e.g `../../`